app/main.py

```diff
@@ -1,34 +1,126 @@
+from dataclasses import dataclass, field
+
+
+@dataclass
 class Deck:
-    def __init__(self, row, column, is_alive=True):
-        pass
 
+    row: int
+    column: int
+    is_alive: bool = True
 
+
+@dataclass
 class Ship:
-    def __init__(self, start, end, is_drowned=False):
-        # Create decks and save them to a list `self.decks`
-        pass
+    start: tuple
+    end: tuple
+    is_drowned: bool = False
+    decks: list = field(init=False)
```

Comment:
The `decks` field is initialized in `__post_init__`, so it should be marked with `init=False` as it is. However, ensure that `default_factory` is not needed here, which is correct in this context.